### PR TITLE
Design Principles For Octopus Server Clients

### PR DESCRIPTION
**Pull request by pawelpabich**
I decided to replace `Revisions` section with `Data Sources` section.  I think that the vast majority of ADR consumers want to know what the current decision is. 

There will be a much smaller group of people who want to gain a deeper understanding of why the decision was made the way it was. `Options & Considerations` and `Data Sources` sections should be able to provide that detailed information.

### CODE REVIEW

File: Decisions/20200716-ClientDesignPrinciples.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ - **Decision owner**: School Of Architecture
+
+## Executive Summary
+
+ Clients should be as light as possible and the heavy lifting should be done by the server. 
+
+## Decision
+
+Keep the logic in the server. 
+
```
Comment (andrewabest) [inline Decisions/20200716-ClientDesignPrinciples.md:14]: I'm wondering if we can format the below like this:

```
The only logic clients should be responsible for is:
- marshalling and sanitising user provided parameters
- allowing users to specify resource parameters by name or Id
  - If references to existing resources need to be passed to a command as one of its inputs then there should be a way to pass them either by `Id` or `Name`. Referring to resources by name might lead long term to a less stable solution but it helps our customers to get a working solution quickly.  
- reporting errors
  - Any errors should be reported back with enough details so the customer knows which inputs need to be modified. 
```

File: Decisions/20200716-ClientDesignPrinciples.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ - **Decision contributors**: Pawel Pabich, Shannon Lewis, Rob Wagner, Mike Noonan
+ - **Decision maker**: TBD
+ - **Decision owner**: School Of Architecture
+
+## Executive Summary
+
+ Clients should be as light as possible and the heavy lifting should be done by the server. 
+
+## Decision
+
```
Comment (andrewabest) [inline Decisions/20200716-ClientDesignPrinciples.md:12]: Could we make this a bit more explicit? Something like "Keep all domain logic in the server".

File: Decisions/20200716-ClientDesignPrinciples.md (hunk 1 above)
Comment (pawelpabich) [reply Decisions/20200716-ClientDesignPrinciples.md:14]: 👍

File: Decisions/20200716-ClientDesignPrinciples.md (hunk 2 above)
Comment (pawelpabich) [reply Decisions/20200716-ClientDesignPrinciples.md:12]: 👍

**Comment by andrewabest** [comment]
:shipit:

**Pull request merged**